src/Weather_forecast_on_dashboard.py

```diff
@@ -0,0 +1,133 @@
+import streamlit as st
+import pandas as pd
+import plotly.express as px
+from herbie import FastHerbie
+from datetime import datetime, timedelta, time
+import numpy as np
+from concurrent.futures import ThreadPoolExecutor
+
+@st.cache_data(show_spinner=False)
+def fetch_data_for_init_time(init_time, forecast_date, lat, lon, parameter, model="ifs"):
+    # Adjust the initialization datetime based on the init_time offset in hours
+    init_datetime = forecast_date - timedelta(hours=init_time)  # This will be a datetime object
+
+    FH = FastHerbie([init_datetime], model=model, fxx=range(0, 24, 1), fast=True)
```

Comment:
I thought you were gona try fxx=range(init_time, 24 + init_time, 1) to help cover all the whole forecast day